src/integrationTests/async.spec.ts

```diff
@@ -49,8 +56,29 @@ describe('when getAdvertisingTokenAsync is called before init', () => {
     });
   });
 
-  describe('when auto refresh fails, but identity still valid', () => {
-    test('it should reject promise after invoking the callback', () => {
+  describe("when initalising with a non-expired identity which requires a refresh", () => {
+    test("it should resolve updated advertising", () => {
+      const originalIdentity = makeIdentity({
+        refresh_from: Date.now() - 100000,
+      });
+      const updatedIdentity = makeIdentity({
+        advertising_token: "updated_advertising_token",
+      });
+      const p = uid2.getAdvertisingTokenAsync().then((token: any) => {
+        expect(callback).toHaveBeenCalled();
```

Comment:
I don't think anything to do with callback belongs in this test - it's not relevant.